tools/tertiary-analysis/scanpy/scanpy_macros2.xml

```diff
@@ -155,18 +155,117 @@ EMBL-EBI https://www.ebi.ac.uk/ and Teichmann Lab at Wellcome Sanger Institute.
+    <data name="matrix_10x" label="${tool.name} on ${on_string}: 10x matrix">
       <filter>export_mtx</filter>
+      <actions>
+        <conditional name="mtx_compression">
+          <when value="None">
+            <action type="format">
+              <!-- I want to update data "matrix_10x" with format="txt" from_work_dir="matrix.mtx" -->
+            </action>
+          </when>
+          <when value="zip">
+            <action type="format">
+              <!-- I want to update data "matrix_10x" with format="zip" from_work_dir="matrix.mtx.zip" -->
+            </action>
+          </when>
+          <when value="gzip">
+            <action type="format">
+              <!-- I want to update data "matrix_10x" with format="gzip" from_work_dir="matrix.mtx.gz" -->
+            </action>
+          </when>
+          <when value="bz2">
+            <action type="format">
+              <!-- I want to update data "matrix_10x" with format="bz2" from_work_dir="matrix.mtx.bz2" -->
+            </action>
+          </when>
+          <when value="zstd">
+            <action type="format">
+              <!-- I want to update data "matrix_10x" with format="zstd" from_work_dir="matrix.mtx.zst" -->
+            </action>
+          </when>
+        </conditional>
+      </actions>
     </data>
-    <data name="genes_10x" format="tsv" from_work_dir="genes.tsv" label="${tool.name} on ${on_string}: 10x genes">
+    <data name="genes_10x" label="${tool.name} on ${on_string}: 10x genes">
       <filter>export_mtx</filter>
+      <actions>
+        <conditional name="mtx_compression">
+          <when value="None">
+            <action type="format">
+              <!-- I want to update data "genes_10x" with format="tsv" from_work_dir="genes.tsv" -->
+            </action>
+          </when>
+          <when value="zip">
+            <action type="format">
+              <!-- I want to update data "genes_10x" with format="zip" from_work_dir="genes.tsv.zip" -->
+            </action>
+          </when>
+          <when value="gzip">
+            <action type="format">
+              <!-- I want to update data "genes_10x" with format="gzip" from_work_dir="genes.tsv.gz" -->
+            </action>
+          </when>
+          <when value="bz2">
+            <action type="format">
+              <!-- I want to update data "genes_10x" with format="bz2" from_work_dir="genes.tsv.bz2" -->
+            </action>
+          </when>
+          <when value="zstd">
+            <action type="format">
+              <!-- I want to update data "genes_10x" with format="zst" from_work_dir="genes.tsv.zst" -->
+            </action>
+          </when>
+        </conditional>
+      </actions>
     </data>
-    <data name="barcodes_10x" format="tsv" from_work_dir="barcodes.tsv" label="${tool.name} on ${on_string}: 10x barcodes">
+    <data name="barcodes_10x" label="${tool.name} on ${on_string}: 10x barcodes">
```

Comment:
I would have multiple outputs, each attached to a potential value of the format.

Reply:
Done in 85534a62c0b6f2e12381b10ccefe860ec46dc408